src/test/LanguageFunctions.test.ts

```diff
@@ -0,0 +1,194 @@
+    'NAB_AL_Tools.sv-SE.xlf',
+];
+let langFilesUri: vscode.Uri[] = [];
+testFiles.forEach(f => {
+    let fromPath = path.resolve(__dirname, testResourcesPath, f);
+    let toPath = path.resolve(__dirname, testResourcesPath, 'temp', f);
+    fs.copyFileSync(fromPath, toPath);
+    langFilesUri.push(vscode.Uri.file(toPath));
+});
+
+suite("Language Functions Tests", function () {
+
+    test("LoadMatchXlfIntoMap()", function () {
+        /* 
+        *   - Test with Xlf that has [NAB:* ] tokens 
+        *   - Assert matchMap does not contain [NAB: *] tokens
+        */
+        let dom = xmldom.DOMParser;
+        let matchMap = LanguageFunctions.LoadMatchXlfIntoMap(new dom().parseFromString(ALObjectTestLibrary.GetXlfHasNABTokens()),xmlns);
+        assert.notEqual(matchMap.size, 0, 'matchMap.size should not equal 0.');
+        assert.equal(matchMap.size, 1, 'matchMap.size should equal 1.');
+        assert.equal(matchMap.get("No Token")?.values().next().value, "No Token");
+        assert.notEqual(matchMap.get('Has Token')?.values().next().value, '[NAB: SUGGESTION]Has Token');
+    });
+
+    test("Run __RefreshXlfFilesFromGXlf() x2", async function () {
+        /**
+         * Tests:
+         *  - Trans-units has been inserted.
+         *  - Trans-units has been removed.
+         */
+        let useExternalTranslationTool = false;
+        let useMatching = true;
+        let sortOnly = false;
+        
+        let refreshResult1 = await LanguageFunctions.__RefreshXlfFilesFromGXlf(gXlfUri, langFilesUri, useExternalTranslationTool, useMatching, sortOnly);
+        assert.equal(refreshResult1.NumberOfAddedTransUnitElements, 26, 'NumberOfAddedTransUnitElements should equal 29.'); // 1. trans-units has been inserted
+        assert.equal(refreshResult1.NumberOfCheckedFiles, langFilesUri.length, 'NumberOfCheckedFiles should equal the length of langFiles[].');
+        assert.equal(refreshResult1.NumberOfRemovedTransUnits, 0, 'NumberOfRemovedTransUnits should equal 0.');
+        assert.equal(refreshResult1.NumberOfUpdatedMaxWidths, 0, 'NumberOfUpdatedMaxWidths should equal 0.');
+        assert.equal(refreshResult1.NumberOfUpdatedNotes, 0, 'NumberOfUpdatedNotes should equal 0.');
+        assert.equal(refreshResult1.NumberOfUpdatedSources, 4, 'NumberOfUpdatedSources should equal 8.'); // 2. trans-units has been removed
+
+        // The function so nice you test it twice
+        let refreshResult2 = await LanguageFunctions.__RefreshXlfFilesFromGXlf(gXlfUri, langFilesUri, useExternalTranslationTool, useMatching, sortOnly);
+        assert.equal(refreshResult2.NumberOfAddedTransUnitElements, 0, 'No new trans-units should have been inserted.');
+        assert.equal(refreshResult2.NumberOfCheckedFiles, refreshResult1.NumberOfCheckedFiles, 'NumberOfCheckedFiles should be the same as last run.');
+        assert.equal(refreshResult2.NumberOfRemovedTransUnits, 0, 'NumberOfRemovedTransUnits should not be equal to 0.');//TODO: Is this correct
```

Comment:
TODO?

Reply:
That was actually meant to be a question. Behaviour was corrected with bug fix in bee4f1c542f111ebcc1e506752fa7abccfc9114a. Since we're running `__RefreshFilesFromGXlf` twice. The lines with "TODO" should be 0 right?

Reply:
Improved in 255a0a86f5e9c002b0a79bb461ce043b7fc7ad1b